### ISSUE TITLE
Error loading Publisher Settings in Superdesk

### ISSUE THREAD

**Issue by alejandrogj** [superdesk/web-publisher#1100]
Hi!

I have linked Superdesk with Publisher but when i click "Publisher Settings" in Superdesk i get a never-ending loading screen. The console prints out this error

`Possibly unhandled rejection: {"data":null,"status":-1,"config":{"method":"POST","transformRequest":[null],"transformResponse":[null],"jsonpCallbackParam":"callback","url":"http://192.168.0.165/api/v2/auth/superdesk/","data":{"session_id":"5eaa70bb30f3daa24f877668","token":"Basic MTI5YzQwNGItNzUwNS00YjEyLWI3MDYtMTUzMTIzODM3NGIzOg=="},"add_format":"json","headers":{"Authorization":"Basic undefined","Accept":"application/json, text/plain, */*","Content-Type":"application/json;charset=utf-8"}},"statusText":"","xhrStatus":"error"}`

It seems there is something wrong whent trying to access to http://192.168.0.165/api/v2/auth/superdesk/

Publisher is installed and working in http://192.168.0.165:8000

My superdesk.config.js 

```
module.exports = function(grunt) {
    return {
        defaultRoute: '/workspace/personal',

        view: {
            timeformat: 'HH:mm',
            dateformat: 'DD.MM.YYYY'
        },

        features: {
            preview: 1,
            swimlane: {columnsLimit: 4},
            noTakes: true,
            editor3: true,
            validatePointOfInterestForImages: false,
            editorHighlights: true,
            noPublishOnAuthoringDesk: true,
            noMissingLink: true
        },
	apps: ['superdesk-publisher'],
	importApps: ['superdesk-publisher'],
	publisher: {
        	protocol: 'http',                /* http or https */
        	tenant: '',                      /* tenant - semantically subdomain, '' is allowed */
 	       domain: '192.168.0.165',           /* domain name for the publisher */
 	       base: 'api/v2',                  /* api base path */

 	       wsProtocol: 'wss',                /* ws or wss (websocket); if unspecified or '' defaults to 'wss' */
 	       wsDomain: '192.168.0.165',  /* domain name (usually domain as above) */
                                                                          /* e.g.: example.com, abc.example.com */
                                                                          /* tenant, as above, is NOT used for websocket */
 	       wsPath: '/ws',                    /* path to websocket root dir */
 	       wsPort: '8080'                    /* if not specified: defaults to 443 for wss, 80 for ws */
	},
    };
};
```
and the .env.local file in Publisher 

```
SUPERDESK_SERVERS='["192.168.0.164"]'
CORS_ALLOW_ORIGIN=http://192.168.0.164
```
Thanks!

**Comment by ahilles107** [comment]
Hey @alejandrogj  `192.168.0.164` is your superdesk server ip? It works without port?

**Comment by alejandrogj** [comment]
Yes, `192.168.0.164` is the superdesk server ip. It works without port.

I installed superdesk using the installing script (https://raw.githubusercontent.com/superdesk/fireq/files/superdesk/install)

**Comment by ahilles107** [comment]
@takeit - any idea here?

**Comment by mabastaraz** [comment]
I'm facing the same issue here, any idea?

**Comment by alejandrogj** [comment]
@mabastaraz not yet :(

**Comment by NicholasIanni** [comment]
Try assigning a hostname to the servers and then use the hostname instead of IP. I had issues caused because of that. Also, did you have any trouble getting the Publisher tab to come up in Superdesk? If so, what guide did you use? I cannot seem to get it to show up...

**Comment by mabastaraz** [comment]
@NicholasIanni Even assigning a hostname to the servers and then use the hostname instead of IP won't solve the problem of the loading screen.

I think it's all related to the configuration of superdesk.config.js inside /opt/superdesk/client, idk maybe I'm wrong!

<img width="1440" alt="Screenshot 2020-05-13 at 18 25 05" src="https://user-images.githubusercontent.com/24724080/81884034-14377300-9597-11ea-8f0f-0d98095bd3b4.png">

**Comment by NicholasIanni** [comment]
@mabastaraz Yeah I realized that :/. Honestly more documentation on the config file would be nice. Out of curiosity did you use the updated script that was posted in the repo's issue area to install?

**Comment by mabastaraz** [comment]
@NicholasIanni Yes, indeed. I installed both fresh Superdesk and Publisher today.
but I have found something interesting,i have tried but didn't work. You can take a look and give us feedback.

https://superdesk-publisher.readthedocs.io/en/latest/cookbooks/developers/websocket.html?highlight=websocket#websocket-communication

**Comment by NicholasIanni** [comment]
Actually after quickly viewing my side of the console something is trying to access "example.com". There is also an issue reaching whatever is running on port 5000 so I'll also look into that if I have time tonight. I think I'm going to post a working virtual machine image for publisher and Superdesk once I get them working so people don't have to go through this process for demos and stuff.

**Comment by NicholasIanni** [comment]
So after reading the article you provided and looking into how things are communicating, it seems the issue definitely is surrounding WebSocket. I'm guessing there is a config problem with either the wsProtocol, wsDomain, wsPath, wsPort, or base variable. It seems that even though 8080 is defined as the port, communication is trying to be made over port 5100. I also some reference to api/v2 but I can't seem to find the was or api/v2 directory. I was hoping to deploy SD and Publisher by end of this week but something tells me we are going to have to do some digging here unless an actual dev can lead us in the right direction...Hope this info is of use. I will keep digging in the meantime.

**Comment by ahilles107** [comment]
guys - what do You have set as value in `SUPERDESK_SERVERS` env (in publisher env's)?

**Comment by ahilles107** [comment]
Here are the docs for this part: https://superdesk-publisher.readthedocs.io/en/latest/manual/getting_started/superdesk-superdesk-publisher-setup.html#update-your-publisher-configuration-file

**Comment by mabastaraz** [comment]
@ahilles107 

/var/www/publisher/.env.local

<img width="427" alt=".env.local" src="https://user-images.githubusercontent.com/24724080/82309268-3aa84480-99c3-11ea-8cd7-6c28ba59a366.png">

/opt/superdesk/client/superdesk.config.js
<img width="719" alt="superdesk.config.js" src="https://user-images.githubusercontent.com/24724080/82309274-3b40db00-99c3-11ea-81fb-6fc2733bab1b.png">

**Comment by ahilles107** [comment]
and you did `composer symfony:dump-env prod` (or dev -if you use dev env)? 
What's the response for superdesk auth request (dev tools -> tan network) ?

**Comment by mabastaraz** [comment]
@ahilles107 
composer symfony:dump-env prod

<img width="616" alt="composer symfony" src="https://user-images.githubusercontent.com/24724080/82312346-354cf900-99c7-11ea-8e6c-36d0fe1c8036.png">



superdesk auth request : ws://localhost:5100/

<img width="1440" alt="3" src="https://user-images.githubusercontent.com/24724080/82312504-71805980-99c7-11ea-9631-67bbeee8ca2c.png">
<img width="1440" alt="2" src="https://user-images.githubusercontent.com/24724080/82312497-6decd280-99c7-11ea-8aff-07fc5ae25025.png">

**Comment by ahilles107** [comment]
there mus be more in network tab - keep it open and refresh page. There must be some filed request to publisher.

**Comment by mabastaraz** [comment]
@ahilles107 After refreshing 
<img width="1440" alt="Screenshot 2020-05-19 at 12 00 26" src="https://user-images.githubusercontent.com/24724080/82313290-6f6aca80-99c8-11ea-9743-d29e4f062342.png">

http://52.59.154.98
admin:admin

**Comment by ahilles107** [comment]
@mabastaraz - it's problem with your publisher instance. http://publisher.kurdishtimes.net/ returns 500 error. Check publisher error log and fix issue ;)

**Comment by ahilles107** [comment]
and in cors allow origin env provide that publisher ip address - as you us it instead domain in your browser.

**Comment by mabastaraz** [comment]
@ahilles107 500 Internal Server Error happens after I run  `composer symfony:dump-env prod`

**Comment by ahilles107** [comment]
This not reason but symptoms. Check publisher/php logs to see what's wrong and fix it ;)

**Comment by mabastaraz** [comment]
@ahilles107 
`rm .env.local.php`
<img width="431" alt="publisher" src="https://user-images.githubusercontent.com/24724080/82321155-b19a0900-99d4-11ea-8833-e7f58d2a1291.png">

when I run `composer symfony:dump-env prod` again
<img width="648" alt="500" src="https://user-images.githubusercontent.com/24724080/82321301-e73ef200-99d4-11ea-9686-ecce44d8d099.png">

**Comment by ahilles107** [comment]
Removing computed env's will not help.  Like i said - check logs and find out real problem.

**Comment by NicholasIanni** [comment]
Where are these logs that you are referencing located?

**Comment by ahilles107** [comment]
Publisher logs can be found at `var/log/` directory inside publisher.

**Comment by mabastaraz** [comment]
@ahilles107 
I'm still facing the same issue after I run `composer symfony:dump-env prod` and .env.local.php generated, the Sperdesk Publisher will go 500 error.

I have checked publisher logs

`root@publisher:/var/log/postgresql# tail postgresql-10-main.log
2020-06-11 18:30:47.684 UTC [31101] root@publisher_prod FATAL: database "publisher_prod" does not exist
2020-06-11 18:30:47.728 UTC [31102] root@publisher_prod FATAL: database "publisher_prod" does not exist
2020-06-11 18:30:47.742 UTC [31103] root@publisher_prod FATAL: database "publisher_prod" does not exist`

Just to make sure that i have put it right information inside /var/www/publisher/.env.local

```
SUPERDESK_SERVERS='["IP/Domian of 
<img width="554" alt="1" src="https://user-images.githubusercontent.com/24724080/84530041-c3748080-ace2-11ea-9e17-1f471bb593fd.png">
Superdesk"]'   ///  IP/Domian of Superdesk
CORS_ALLOW_ORIGIN=http://IP/Domian of Superdesk Publisher  /// IP/Domian of Superdesk Publisher
```


Also included a screenshot of .env.local.php



Thanks for your help :)
